### DISCUSSION TITLE
Does Rotki support sharing my data with friends in read-only mode?

### DISCUSSION THREAD

**Discussion by WinPoss** [rotki/rotki#7970] [Q&A]
Does Rotki support sharing my data with friends in read-only mode? For example, can I generate a link and share it with my friends, so that when they open the link, they can view my investment data on the webpage?

Thank you.

**Comment by yabirgb** [comment]
Hello! For this kind of questions is better to use [discussions](https://github.com/rotki/rotki/discussions) or discord.rotki.com 

At the moment we don't have such functionality. What you can do is share your database and global.db along the password. The password can be changed from the settings. May I ask what kind of data you would like to share? Maybe we could design something that could be exported as summary

**Comment by WinPoss** [comment]
> Hello! For this kind of questions is better to use [discussions](https://github.com/rotki/rotki/discussions) or discord.rotki.com
> 
> At the moment we don't have such functionality. What you can do is share your database and global.db along the password. The password can be changed from the settings. May I ask what kind of data you would like to share? Maybe we could design something that could be exported as summary



I would like to generate a read-only public page that allows users to showcase their investment performance. Users can share this page with friends, communities, or social networks, thereby driving new visitors to the platform.

**The page should display the following five sections of data:**

1. Performance curve of the exchange account, with options to display curves for 7 days, 30 days, 180 days, and 365 days, and all time.
2. Initial assets, final assets, and overall return rate of the account within the selected time period.
3. Within the selected time period, the account's overall win rate, profit/loss ratio, Sharpe Ratio, maximum drawdown, number of winning positions, and total number of positions. 
4. Current holdings of the account, displaying the quantity and returns of each position.
5. Historical trade details of the account within the selected time period, including buy/sell transactions, profit/loss amount, and return rate for each trade.

**Users should have the option to choose whether to display sections 4 and 5.**

The benefit of this feature is that it allows users to share the link, thereby driving traffic to the website. I hope you can consider adding this functionality.

**Comment by WinPoss** [comment]
Additionally, I would like to ask if it is possible to deploy the Rotki project on a Linux server as a website for users to use? Thank you.

**Comment by LefterisJP** [comment]
I am going to convert this to a discussion since this is not really an issue requests

**Comment by LefterisJP** [comment]
So thanks for the questions @winposs!

> I would like to generate a read-only public page 

So I am not sure this is possible for a local application. Since it's a self-hosted application you can't really share anything since you would need to somehow create a server open it to the internet and display the data.

If I properly understand the motivation behind it is so that we can market rotki better by a feature of publicly sharing a snapshot of the user's data, right? Serving as both a way to get more people to use the platform and also as a way to share data publically.

It could be a completely centralized way of doing it by using our server. Without any privacy since it's public but it may not be a terrible idea. Is this what you mean?

> Additionally, I would like to ask if it is possible to deploy the Rotki project on a Linux server as a website for users to use? Thank you.

rotki can run on docker, it can run on linux also. But it's not a multiuser application. You can have multiple users, but all belonging to same entity as some common data are shared and it would not be safe to have different users in the same system that belong to different entities.

**Reply by WinPoss** [reply]
Sorry, I originally understood that rotki is a web application. If it is a local application, my suggestion may not be suitable.
